main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/goccy/go-json"
+	"github.com/klauspost/compress/gzip"
+)
+
+const (
+	serverTimeout = time.Minute
+)
+
+var config struct {
+	serverPort         string
+	failAuthChance     float64
+	slowReceiveChance  float64
+	badResponseChance  float64
+	randomNetworkLag   int
+	disableJsonParsing bool
+}
+
+func main() {
+	parseConfig()
+
+	log.Println("starting the server at port " + config.serverPort)
+	srv := http.Server{
+		Addr:         ":" + config.serverPort,
+		Handler:      getRouter(),
+		ReadTimeout:  serverTimeout,
+		WriteTimeout: serverTimeout,
+	}
+
+	panic(srv.ListenAndServe())
+}
+
+func parseConfig() {
+	flag.StringVar(&config.serverPort, "port", "8083", "local port to launch the server at")
+	flag.Float64Var(&config.failAuthChance, "random_no_auth", 0, "chance to fail authentication, from 0.0 to 1.0")
+	flag.Float64Var(&config.slowReceiveChance, "random_slow_receive", 0, "chance to receive data slowly, from 0.0 to 1.0")
+	flag.Float64Var(&config.badResponseChance, "random_bad_response", 0, "chance to return a non-200 response status code, from 0.0 to 1.0")
+	flag.IntVar(&config.randomNetworkLag, "random_network_lag", 0, "maximum random network lag on receiving request and responding, msec")
+	flag.BoolVar(&config.disableJsonParsing, "disable_json_parsing", false, "disable payload unmarshalling to increase processing speed, boolean")
+	flag.Parse()
+}
+
+func getRouter() *chi.Mux {
+	mux := chi.NewMux()
+
+	mux.Use(middleware.StripSlashes)
+	mux.Use(randomFailMiddleware)
+
+	mux.Post("/", handleLogs)
```

Comment:
Maybe it would make sense to follow the Datadog API paths already from the start. We can anticipate adding some more endpoints and then it would be a hassle to have to break the API.

https://docs.datadoghq.com/api/latest/logs/